src/lib/Libifl/pbsD_confirmresv.c

```diff
@@ -64,7 +64,7 @@
  */
 
 int
-pbs_confirmresv(int c, char *rid, char *location, unsigned long start,
+__pbs_confirmresv(int c, char *rid, char *location, unsigned long start,
```

Comment:
Why are you using the double underscore prefix? You've defined a macro for pbs_confirmresv().

Reply:
Other IFL function implementations follow this syntax as well. The macros are set to the dereferenced value of the function pointers, which in fact point to the IFL function implementations, and those functions have the double underscore prefix. That way, somebody could define their own pbs_confirmresv and call it, say, 'my_confirmresv', and then all that they need to do is point the function pointer pfn_pbs_confirmresv to my_confirmresv and all calls to pbs_confirmresv will automatically call my_confirmresv.

Reply:
So if I wanted to write my own version of pbs_confirmresv() then I could call it __custom_confirmresv() and redefine the macro (either in the header or on the command line) such that pbs_confirmresv() references my function rather than __pbs_confirmresv(). For that reason, I don't understand why the double underscore prefix is used here. Also, should the macros in pbs_ifl.h each be defined only when they are undefined? For example...
```
#ifndef pbs_confirmresv
#define pbs_confirmresv(a, b, c, d, e)	(*pfn_pbs_confirmresv)(a, b, c, d, e)
#endif
```

You could also do the following to avoid having to list the arguments...
```
#ifndef pbs_confirmresv
#define pbs_confirmresv	(*pfn_pbs_confirmresv)
#endif
```

Reply:
I think double underscore is just something that is used to define internal implementations and underscores were added to avoid collisions with user's implementation. As far as I know, this is a usual practice followed in C++. I think it wouldn't create any problems as such.

Reply:
I understand the convention of using double underscores. And I also just realized that I'm looking at the definition, rather than a call to the function. I'm barking up the wrong tree. Sorry about that.